crates/flux-middle/src/queries.rs

```diff
@@ -583,6 +587,24 @@ impl<'genv, 'tcx> Queries<'genv, 'tcx> {
         })
     }
 
+    pub(crate) fn no_panic(&self, genv: GlobalEnv, def_id: DefId) -> QueryResult<bool> {
```

Comment:
This is infallible, so we don't need to return a result here.

```suggestion
    pub(crate) fn no_panic(&self, genv: GlobalEnv, def_id: DefId) -> bool {
```